Specify version 0.4.5 of gdbstub

AVassilev98 opened the pull request.

daniel5151 commented:
You'll also want to run `cargo update` to update the version number in `Cargo.lock`

AVassilev98 commented:
I picked 0.4.1 because of the comment above `# this should be ">=0.4.1", but I haven't released 0.4.1 yet...`. Updated to 0.4.5 and removed the comment.

Also good point on the `cargo update`, my apologies for that, I am completely unfamiliar with Rust development (too busy fixing c++ memory bugs to learn ;) )

daniel5151 commented:
Nice, LGTM.

> I am completely unfamiliar with Rust development (too busy fixing c++ memory bugs to learn ;) )

Haha, all good!

FWIW, if I could redo CS 452 again (and wasn't allowed to use Rust for my OS), I'd have stuck to plain 'ol C instead of using C++ for the OS. You're not getting all that much using C++ in such a simple kernel, and the added abstraction and complexity only made debugging things harder for my partner and I.

...or maybe it's because we used a bunch of `std::optional`, `std::variant`, and `std::span`, all of which are just terrible constructs, and in true C++ fashion, come with _no_ effective guardrails against footguns whatsoever.
